File: sqlite-android-studio-plugin/src/main/java/com.alecstrong.sqlite.android/psi/ASTWrapperPsiElement.java
package com.alecstrong.sqlite.android.psi;

import com.intellij.lang.ASTNode;
import com.intellij.psi.PsiElement;
import com.intellij.psi.util.PsiUtilCore;
import java.util.ArrayList;
import java.util.List;
import org.jetbrains.annotations.NotNull;

public class ASTWrapperPsiElement extends com.intellij.extapi.psi.ASTWrapperPsiElement {
  public ASTWrapperPsiElement(@NotNull ASTNode node) {
    super(node);
  }

  @NotNull @Override public PsiElement[] getChildren() {
    PsiElement psiChild = getFirstChild();
    if (psiChild == null) return PsiElement.EMPTY_ARRAY;

    List<PsiElement> result = new ArrayList<PsiElement>();
    while (psiChild != null) {
      result.add(psiChild);
      psiChild = psiChild.getNextSibling();
    }
    return PsiUtilCore.toPsiElementArray(result);
  }
}

File: sqlite-android-studio-plugin/src/main/kotlin/com/alecstrong/sqlite/android/psi/ASTWrapperPsiElement.kt
package com.alecstrong.sqlite.android.psi

import com.intellij.lang.ASTNode
import com.intellij.psi.PsiElement
import com.intellij.psi.util.PsiUtilCore
import java.util.ArrayList

open class ASTWrapperPsiElement(node: ASTNode) : com.intellij.extapi.psi.ASTWrapperPsiElement(
    node) {
  override fun getChildren(): Array<PsiElement> {
    var psiChild: PsiElement? = firstChild ?: return PsiElement.EMPTY_ARRAY

    val result = ArrayList<PsiElement>()
    while (psiChild != null) {
      result.add(psiChild)
      psiChild = psiChild.nextSibling
    }
    return PsiUtilCore.toPsiElementArray(result)
  }
}
